src/components/instructor/AssignmentAdd.js

```diff
@@ -1,18 +1,97 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import { DemoContainer } from '@mui/x-date-pickers/internals/demo';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import dayjs from 'dayjs';
 
-// complete the code.  
+// complete the code.
 // instructor adds an assignment to a section
 // use mui Dialog with assignment fields Title and DueDate
 // issue a POST using URL /assignments to add the assignment
 
 const AssignmentAdd = (props)  => {
+    const [open, setOpen] = useState(false);
+    const [message, setMessage] = useState('');
+    const [newTitle, setNewTitle] = useState('');
+    const [newDate, setNewDate] = useState('');
 
-   
+    const [newAssignment, setNewAssignment] = useState(null);
+
+    useEffect(() => {
+        // don't do anything if it's first time
+        if (newAssignment !== null) {
+            props.save(newAssignment);
```

Comment:
For reviewers:
`props.save()` points to the `addAssignment` function in `AssignmentView.js`